source/style-guide/style/code-examples.txt

```diff
@@ -29,21 +30,20 @@ or output examples:
 -  When the command is simple, and there's nothing specific to say
    about the output, you can show the input and output in the same code
    block, as users would actually see the code in their own terminal.
-   The inclusion of the command prompt differentiates the input from
-   the output.
+   In this case, you can include the command prompt to differentiate the
+   input from the output. Ensure that this code block is set to 
+   ``:copyable: false``.
 
 -  Ensure that any placeholder text in code is obvious.
 
-   - If the authoring tool allows it, apply italics to placeholders; if
-     not, enclose them in angle brackets.
+   - Enclose placeholders in angle brackets.
    - Use lowercase letters for single-word placeholders. To show
-     multiple-word placeholders, don't separate the words with spaces
-     or symbols and capitalize the first letter of each word after the
-     first word (camelCase).
+     multiple-word placeholders, user lowercase letters and separate the
+     words with hyphens (kebab-case).
```

Comment:
Open Q. We do this differently in different repos and need to discuss.

Reply:
**advice:**

This depends on the language and type of variable.

- Shell (including Windows PowerShell and Command Prompt): `<variable-name>` Use and link to [docopt](http://docopt.org/)
- API: `{variableName}` (See [OpenAPI spec](https://swagger.io/docs/specification/describing-parameters/).)
- Java: `<variable_name>`, `<class_data_member_>` : Use and link to [Google's Java style guide](https://google.github.io/styleguide/javaguide.html#s5-naming)
- Scala: `<variableName>` : Use and link to [Scala's naming conventions](https://docs.scala-lang.org/style/naming-conventions.html)
- C++: Use and link to [Google's C++ style guide](https://google.github.io/styleguide/cppguide.html#Naming)
- C#: `<variableName>` : Use and link to [Microsoft's C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions#naming-conventions)
- Python: `<variable_name>` : Use and link to [PEP 8: Style Guide for Python](https://www.python.org/dev/peps/pep-0008/#id37)
- Ruby: `<variable_name>` : Use and link to [Ruby doc style guide](https://github.com/rubocop-hq/ruby-style-guide)
- PHP: `<$variable>` : Use and link to [PSR-12](https://www.php-fig.org/psr/psr-12/)

Reply:
**suggestion:** We may want to set some further standards as to whether or not we use placeholders in certain circumstances. @ccho-mongodb or @cbush may have thoughts here.

Reply:
Is a placeholder a variable name, a portion of a string that should be replaced, or a parameter/argument? E.g.:
1) `const myPlaceholder = 4`
or
2) `const uri = "https://<myConnectionString>?retryWrites=true"`
or
3) `grep <yourRegularExpression>`

Reply:
Would also be great to standardize commonly-used placeholders. Do we also want to standardize "placeholder text" capitalization in which we provide examples? For example, we could standardize our connection uri string placeholders and examples. On the server manual page https://docs.mongodb.com/manual/reference/connection-string/ , there are several variations:
- `mongodb://[username:password@]host1[:port1][,...hostN[:portN]][/[defaultauthdb][?options]]`
- `mongodb://myDBReader:D1fficultP%40ssw0rd@mongos0.example.com:27017,mongos1.example.com:27017,mongos2.example.com:27017/?authSource=admin`
- `mongodb+srv://server.example.com/?connectTimeoutMS=300000&authSource=aDifferentAuthDB`
- `mongodb://mongodb1.example.com:27317,mongodb2.example.com:27017/?connectTimeoutMS=300000&replicaSet=mySet&authSource=aDifferentAuthDB`

Throughout the new driver documentation, we use:
`const uri =
  "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";`

Reply:
@ccho-mongodb : docopt covers a lot of that. I think language conventions are important, too.